TreeMaker/interface/TreeMaker.h

```diff
@@ -356,3 +358,83 @@ class TreeRecoCand : public TreeObject<vector<TLorentzVector> > {
 		bool doLorentz{};
 		vector<double> pt, eta, phi, energy;
 };
+
+//derived version of vector<vector<T>>
+//with switch for vector<T> values and vector<int> offsets instead
+template <template<typename...> class R=std::vector, typename Top = R<R<double>>, typename Sub = typename Top::value_type>
```

Comment:
in what case would R not be std::vector?

Reply:
wouldn't we prefer to just make the inner type a template, and then build up Sub and Top from that?

Reply:
1) I guess in our case we should restrict R to be std::vector. I took this code from somewhere else and if you look at the flatten function it technically doesn't require the outer object to be a std::vector. Do you want me to work on removing R?
2) Doing (1) would make this possible, but is it really necessary? Seems to be roughly equivalent code.

Reply:
I think it's preferable to make the code as simple as possible (or in other words, only as complex as necessary). It's simpler to specify one parameter, the inner type, and then just typedef Sub and Top as vectors of that inner type. The use of value_type is clever, but unnecessarily so.

Reply:
Okay, the new commit should make things simpler.